src/tests/support.cpp

```diff
@@ -892,3 +892,79 @@ rnp_tests_key_search(rnp_key_store_t *keyring, const std::string &keyid)
     srch_userid.by.userid[sizeof(srch_userid.by.userid) - 1] = '\0';
     return rnp_key_store_search(keyring, &srch_userid, NULL);
 }
+
+void
+reload_pubring(rnp_ffi_t *ffi)
+{
+    rnp_output_t output = NULL;
+    assert_rnp_success(rnp_output_to_path(&output, "pubring.gpg"));
+    assert_rnp_success(rnp_save_keys(*ffi, "GPG", output, RNP_LOAD_SAVE_PUBLIC_KEYS));
+    assert_rnp_success(rnp_output_destroy(output));
+    assert_rnp_success(rnp_ffi_destroy(*ffi));
+    /* re-init ffi and load keys */
+    assert_rnp_success(rnp_ffi_create(ffi, "GPG", "GPG"));
+    rnp_input_t input = NULL;
+    assert_rnp_success(rnp_input_from_path(&input, "pubring.gpg"));
+    assert_rnp_success(rnp_import_keys(*ffi, input, RNP_LOAD_SAVE_PUBLIC_KEYS, NULL));
+    assert_rnp_success(rnp_input_destroy(input));
+    assert_int_equal(unlink("pubring.gpg"), 0);
+}
+
+void
+reload_keyrings(rnp_ffi_t *ffi)
+{
+    rnp_output_t output = NULL;
+    assert_rnp_success(rnp_output_to_path(&output, "pubring.gpg"));
```

Comment:
I usually prefer to use memory input/output, to not rely on the filesystem, avoid possible file collisions and speed up a bit. Are there any objections against it?

Reply:
@rrrooommmaaa Agree, let's change to the memory input/output.